packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -261,4 +263,17 @@ contract MixinExchangeCore is
             order.takerAssetData
         );
     }
+
+    /// @dev Calculates the sum of values already filled and cancelled for a given order.
+    /// @param orderHash The Keccak-256 hash of the given order.
+    /// @return Sum of values already filled.
+    function getUnavailableTakerTokenAmount(bytes32 orderHash)
+        public view
+        returns (uint256)
+    {
+        if (cancelled[orderHash]) {
```

Comment:
An order is also cancelled if its salt less than the maker's `makerEpoch`. There is a `getOrderStatus` in an one of my upcoming PR's that could be useful here.